lib/phoenix_component.ex

```diff
@@ -1422,17 +1423,24 @@ defmodule Phoenix.Component do
   def to_form(data, options \\ [])
 
   def to_form(%Phoenix.HTML.Form{} = data, options) do
-    case Keyword.fetch(options, :as) do
-      {:ok, as} ->
-        name = if as == nil, do: as, else: to_string(as)
-        %{data | name: name, id: Keyword.get(options, :id) || name}
-
-      :error ->
-        case Keyword.fetch(options, :id) do
-          {:ok, id} -> %{data | id: id}
-          :error -> data
-        end
-    end
+    {name, id} =
+      case Keyword.fetch(options, :as) do
+        {:ok, as} ->
+          name = if as == nil, do: as, else: to_string(as)
+          {name, Keyword.get(options, :id) || name}
+
+        :error ->
+          case Keyword.fetch(options, :id) do
+            {:ok, id} -> {data.name, id}
+            :error -> {data.name, data.id}
+          end
+      end
+
+    {_as, options} = Keyword.pop(options, :as)
+    {errors, options} = Keyword.pop(options, :errors, data.errors)
```

Comment:
I think we can skip supporting errors. What are your thoughts?

Reply:
I liked the idea of being able to overwrite errors via `to_form/2` but we don't have to keep it.

We _do_ have to at least ignore errors though, for instance:

```elixir
{_errors, options} = Keyword.pop(options, :errors)
```

...otherwise if errors are passed to `<.form for={@form} errors={[name: "err"]}>` we raise with a message about how lists can only contain binaries.

Reply:
I was cribbing from the tests, so I thought it was something we would necessarily encounter :)

https://github.com/phoenixframework/phoenix_live_view/blob/d90f6915fdb4fa020c9f82f55e9bbf89e60b8a1b/test/phoenix_component/components_test.exs#L406-L426

After more thoughts, skipping `:errors` with prebuilt forms will lead to less confusion. I was thinking about the case where you want to apply errors to an existing `Form.t` via `to_form/2`, but the clause that accepts Form is really just to drive the form component, not to manage the Form itself.

I will update to pop the errors! :)

Reply:
lol reverting